lib/plugindata/cas.go

```diff
@@ -0,0 +1,129 @@
+package plugindata
+
+import (
+	"context"
+	"time"
+
+	"github.com/gravitational/teleport-plugins/lib/backoff"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+)
+
+const (
+	// backoffBase is an initial (minimum) backoff value.
+	backoffBase = time.Millisecond
+	// backoffMax is a backoff threshold
+	backoffMax = time.Second
+)
+
+// Client represents an interface to Teleport API client
+type Client interface {
+	GetPluginData(context.Context, types.PluginDataFilter) ([]types.PluginData, error)
+	UpdatePluginData(context.Context, types.PluginDataUpdateParams) error
+}
+
+// EncodeFn represents a function type of plugin data serializer
+type EncodeFn = func(interface{}) map[string]string
+
+// DecodeFn represents a function type of plugin data deserializer
+type DecodeFn = func(map[string]string) interface{}
+
+// Cb represents a type for Plugin Data modification fn
+type Cb = func(interface{}) (interface{}, bool)
```

Comment:
Can we rename this to something more descriptive? Does this stand for "callback"?